### PR TITLE
signed and unsigned tests

### PR DESCRIPTION
**Pull request by found101**
I erased the reporting module from my fork because it started causing errors in my tests and I only wanted to add working code upstream. The reporting test module was really small and can easily be recreated later. It currently is not upstream either.

### CODE REVIEW

File: src/tests/system_tests.rs [hunk 1]
```diff
@@ -51,6 +51,8 @@ fn test_example_file(file: &str) -> bool {
 #[test]
 fn test_examples() {
     assert!(test_example_file("test_conditions"));
+    assert!(test_example_file("test_signed"));
```
Comment (arc3x) [inline src/tests/system_tests.rs]: these were supposed to go in the commented functions below

File: tests/test_signed.rs [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+// FIXME: some preconditions are unnecisarilly restricitve ie add_five_i32_invalid does not need x: i32 >= i32::MIN + 5i32
+
+// * * *
+// Integer Add Tests
+// * * *
+
+// Tests signed integer overflow for 32 bit integers
+// Should be invalid
+#[condition(pre="(x: i32 > 0) && (x: i32 < 10)", post="return: (x: i32 > 30)")]
```
Comment (arc3x) [inline tests/test_signed.rs]: the postcondition `return: (x: i32 > 30)` doesn't make sense to me.
the format is `<variable>:<type> <operator> <value>`
to me it seems as if you are describing returns type as `(x: i32 > 30)`
I am surprised this doesn't cause a parsing error.
@salterm should take a look at this. I suspect it is because our parser is more permissive than it needs to be. I think this might have been a design decision at some point for some reason.

Also, this test doesn't test for signed integer overflow like the comment implies. In your precondition you restrict x to greater than 0 && less than 10; any value of [1..9] plus or minus five can fit in an i32.

For example to test for overflow I would be inclined to write:
```
// Tests signed integer overflow catching for 32 bit integers
// Should be invalid
#[condition(pre="x:i32 <= i32::MAX - 4i32", post="return: i32 == (x:i32 + 5i32)")]
fn add_five_i32_invalid(x: i32) -> i32 {
    x+5
}
```

Comment (arc3x) [inline tests/test_signed.rs]: @salterm disreguard. This is caught by the parser, but tests pass regardless. I will fix up the tests.

File: tests/test_signed.rs [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+fn invalid_add_five_i32(x: i32) -> i32 {
+    x+5
+}
+
+// Tests signed integer overflow for 32 bit integers
+// Should be valid
+#[condition(pre="(x: i32 > 0) && (x: i32 < 10)", post="return: (x: i32 > 3)")]
```
Comment (arc3x) [inline tests/test_signed.rs]: all these constants should be type annotated. ie 10i32

**Comment by VSchuster** [comment]
Outdated from the day
